### DISCUSSION TITLE
Channel Layer GROUP issue or mb ?

### DISCUSSION THREAD

**Discussion by Guidoow** [django/channels#1931] [Help!]
Hi and thanks for reading!

I'm experimenting with in memory channel layers, because I can't use Redis with Windows yet, and I ran into a problem when I tried to consume a message from groups.
As I understood from reading the docs, the group_send method adds sent messages to a queue and, if there is a specified "type" key in the sent dictionary, executes a consumer function (named in the key value) that is in the specified group and has a persistent connection.

I'm new to development, anyway I got around to exploring the code, and I can tell that all the messages I send are perfectly queued, but there is no function execution.

Maybe there isn't only a persistent connection required from the user who is in the group, but too a message sent to the receive() consumer method for then manually check for queued messages, read and send them? or maybe I am doing something wrong?

Again, thanks for reading. <3

**Comment by carltongibson** [comment]
Hi @Guidoow — can you show how you're invoking the `group_send()`? i.e. the consumer. 

A **minimal** reproduce would be good to be able to help you

**Reply by Guidoow** [reply]
group_add in consumer connect method: 
`async_to_sync(self.channel_layer.group_add)('chat',self.channel_name)`

group_send when a user sent a message: 
`async_to_sync(self.channel_layer.group_send)('chat',message)`
how i said i check it was stored by a terminal print: 
`<Queue maxsize=0 _queue=[(1665792743.370699, {'userFrom': 'guido', 'fecha': '2022-10-14 21:11:23.293981', 'message': 'gh test', 'image': ''})] tasks=1>`

I also tried creating a single group for each consumer and send messages to himself, with one and two consumers and the same time and no one method works :(

**Reply by carltongibson** [reply]
Hi @Guidoow -- can I ask you to put together a minimal reproduce please? I can't see the issue from just these fragments.

**Reply by Guidoow** [reply]
Hi, i have some things mixed in my code and maybe is it not really clear, but i hope this helps!
main methods receive and connect.
![connect receive](https://user-images.githubusercontent.com/104577925/195994863-9081fd12-d4f2-41aa-b977-1ad46799062c.png)
Auth method and setting channel names.
![authent](https://user-images.githubusercontent.com/104577925/195994864-c5e8a4d6-fcb8-4ede-8c2b-992fde838c02.png)
sending messages:
![group_send](https://user-images.githubusercontent.com/104577925/195994865-44cf49c8-98a8-4825-ad96-0c313cccaddc.png)
function that should execute automatically?
![chat message](https://user-images.githubusercontent.com/104577925/195994867-83d116e8-2996-4e07-aa45-094fc0d97f85.png)
console log:
![console log](https://user-images.githubusercontent.com/104577925/195995502-08a5edae-f6f8-4087-8eee-eff2f3fc5a03.png)


As you can see in the "group_send" image, I was trying to send to my same channel and then receive the messages with a function that executes: _'async_to_sync(self.channel_layer.receive)(self.user)'_  and works! but when I tried to receive and there is no message in the queue, the server stops completely and crashes.

if there is any issue, please consider contacting me via gmail !

**Reply by Guidoow** [reply]
@carltongibson I had to perform a windows 0 boot because it gave me some errors on actualization, fortunately I can use Redis correctly, but I still can't fix the channels error :/
